mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerStorageTest.java

```diff
@@ -534,4 +537,370 @@ public void accessibleMessagesShouldFilterOutMessageIdsWhenNotBelongingToTheUser
         return messageResult -> messageResult.getMailboxId().equals(mailboxId);
     }
 
+    @Test
```

Comment:
I think previous tests on accessibleMessages should be changed too.

Reply:
I don't understand this comment.

I did not change previous test (not involving delegation). Keeping these "old" test unchanged is crucial to me to not introduce vulnerabilities

I also added detailed new test demonstrating **accessibleMessages** in a delegation context. Namely accessibleMessagesShouldReturnMessagesWhenReadDelegated and accessibleMessagesShouldNotReturnMessagesWhenNotReadDelegated